pkgs/epnix/tools/phoebus/client/default.nix

```
@@ -1,112 +1,50 @@
-      (epnix.phoebus-setup-hook.override {
-        jdk = jdk.override {
-          enableJavaFX = true;
-          openjfx = openjfx.override {
-            withWebKit = true;
-          };
-        };
-      })
-    ];
-
-    # Put runtime dependencies in propagated
-    # because references get thrown into a jar
-    # which is compressed,
-    # so the Nix scanner won't always be able to see them
-    propagatedBuildInputs = [
-      python3
-    ];
-
-    desktopItems = [
-      (makeDesktopItem {
-        name = "phoebus";
-        exec = "phoebus -server 4918 -resource %f";
-        desktopName = "Phoebus";
-        keywords = ["epics" "css"];
-        # https://specifications.freedesktop.org/menu-spec/menu-spec-1.0.html#category-registry
-        categories = [
-          # Main
-          "Office"
-
-          # Additional
-          "Java"
-          "Viewer"
-        ];
-      })
-    ];
-
-    buildPhase = ''
-      runHook preBuild
-
-      # Copy deps to a writable directory, due to the usage of "install-jars"
-      local deps=$PWD/deps
-      cp -r --no-preserve=mode "${epnix.phoebus-deps}" $deps
-
-      # TODO: tests fail
-      mvn package \
-        --projects "./phoebus-product" \
-        --also-make \
-        --offline \
-        -Dmaven.javadoc.skip=true -Dmaven.source.skip=true -DskipTests \
-        -Dproject.build.outputTimestamp=${buildDate} \
-        -Dmaven.repo.local="$deps"
-
-      runHook postBuild
-    '';
-
-    installPhase = ''
-      runHook preInstall
-
-      installPhoebusJar \
-        "phoebus-product/" \
-        "product-$version.jar" \
-        "phoebus" \
-        "org.phoebus.product.Launcher"
-
-      # MIME types for PV Tables
-      install -D -m 444 phoebus-product/phoebus.xml -t $out/share/mime/packages
-
-      runHook postInstall
-    '';
-
-    meta = {
-      description = "Control System Studio's Phoebus client";
-      homepage = "https://control-system-studio.readthedocs.io/en/latest/index.html";
-      mainProgram = "phoebus";
-      license = lib.licenses.epl10;
-      maintainers = with epnixLib.maintainers; [minijackson];
-      inherit (jdk.meta) platforms;
-    };
-  }
+  stdenv,
+  # Inspired by:
+  # https://epics.anl.gov/tech-talk/2024/msg00895.php
+  java_opts ? "-XX:MinHeapSize=128m -XX:MaxHeapSize=4g -XX:InitialHeapSize=1g -XX:MaxHeapFreeRatio=10 -XX:MinHeapFreeRatio=5 -XX:-ShrinkHeapInSteps -XX:NativeMemoryTracking=detail",
+}:
+stdenv.mkDerivation {
+  pname = "phoebus";
+  inherit (epnix.phoebus-unwrapped) version;
+  nativeBuildInputs = [makeWrapper copyDesktopItems];
+
+  dontUnpack = true;
+  dontBuild = true;
+  dontConfigure = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    # This wrapper for the `phoebus-unwrapped` executable sets the `JAVA_OPTS`
+    makeWrapper "${lib.getExe epnix.phoebus-unwrapped}" "$out/bin/$pname" \
+      --prefix JAVA_OPTS ":" "${java_opts}"
```

`JAVA_OPTS` is a variable separated by spaces (instead of `:`). Found this by looking at the generated wrapper script, and saw that it weirdly split the `${java_opts}` by `:`. I don't know if this could lead to issues, but let's not take that risk.

```suggestion
      --prefix JAVA_OPTS " " "${java_opts}"
```